sockeye/inference.py

```diff
@@ -44,32 +44,28 @@ class InferenceModel(model.SockeyeModel):
     :param beam_size: Beam size.
     :param checkpoint: Checkpoint to load. If None, finds best parameters in model_folder.
     :param softmax_temperature: Optional parameter to control steepness of softmax distribution.
+    :param max_output_length_num_stds: Number of standard deviations as safety margin for maximum output length.
     """
 
     def __init__(self,
                  model_folder: str,
                  context: mx.context.Context,
                  fused: bool,
-                 max_input_len: Optional[int],
                  beam_size: int,
                  checkpoint: Optional[int] = None,
-                 softmax_temperature: Optional[float] = None):
+                 softmax_temperature: Optional[float] = None,
+                 max_output_length_num_stds: int = C.DEFAULT_NUM_STD_MAX_OUTPUT_LENGTH,):
         self.model_version = utils.load_version(os.path.join(model_folder, C.VERSION_NAME))
         logger.info("Model version: %s", self.model_version)
         utils.check_version(self.model_version)
 
-        # load config & determine parameter file
         config = model.SockeyeModel.load_config(os.path.join(model_folder, C.CONFIG_NAME))
-        if max_input_len is None:
-            max_input_len = config.max_seq_len_source
-        else:
-            if max_input_len != config.max_seq_len_source:
-                logger.warning("Model was trained with max_seq_len_source=%d, but using max_input_len=%d.",
-                               config.max_seq_len_source, max_input_len)
-        config.max_seq_len_source = max_input_len
         super().__init__(config)
 
-        fname_params = os.path.join(model_folder, C.PARAMS_NAME % checkpoint if checkpoint else C.PARAMS_BEST_NAME)
+        self.max_input_length = config.max_seq_len_source
```

Comment:
thanks for the change! Having this as an attribute rather than overwriting the config looks cleaner :)